src/main/java/com/minecolonies/coremod/commands/RefreshColonyCommand.java

```diff
@@ -0,0 +1,137 @@
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static com.minecolonies.coremod.commands.AbstractSingleCommand.Commands.REFRESH_COLONY;
+import static com.minecolonies.coremod.commands.AbstractSingleCommand.Commands.SHOWCOLONYINFO;
+
+/**
+ * List all colonies.
+ */
+public class RefreshColonyCommand extends AbstractSingleCommand
+{
+
+    public static final  String DESC                       = "refresh";
+    private static final String NO_COLONY_FOUND_MESSAGE_ID = "Colony with ID %d not found.";
+    private static final String NO_ARGUMENTS               = "Please define a colony to refresh";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public RefreshColonyCommand(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId|OwnerName>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        final int colonyId;
+        if(args.length == 0)
+        {
+            IColony colony = null;
+            if(sender instanceof EntityPlayer)
+            {
+                colony = ColonyManager.getIColonyByOwner(((EntityPlayer) sender).worldObj, (EntityPlayer) sender);
+            }
+
+            if(colony == null)
+            {
+                sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NO_ARGUMENTS));
+                return;
+            }
+            colonyId = colony.getID();
+        }
+        else
+        {
+            colonyId = getIthArgument(args, 0, -1);
+        }
+
+        final EntityPlayer player = (EntityPlayer) sender;
+        if (!canPlayerUseCommand (player, Commands.valueOf("REFRESH_COLONY"), colonyId))
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NOT_PERMITTED));
+            return;
+        }
+
+        final Colony colony = ColonyManager.getColony(colonyId);
+
+        if(colony == null)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NO_COLONY_FOUND_MESSAGE_ID));
+            return;
+        }
+
+        if(sender instanceof EntityPlayer)
+        {
+
+            if (!colony.getPermissions().getRank(player).equals(Permissions.Rank.OWNER) ||isPlayerOpped(player, "REFRESH_COLONY"))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)